### PR TITLE
Readds the tesla, reworks it a bit

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

I "borrowed" code from #52930, thanks for making such a neat thing goof.

Re-adds the tesla as a supermatter delam while removing its ability to blow shit up.
Forbids the tesla from damaging doors, maybe this way we won't get 100s of atmos diffs from airlocks to space breaking.
Lets the tesla damage structures so it can't get stuck in place nearly as easily.
Forbids portable atmos machines from being zapped, as they don't do anything to them.
Uses the zapped targets list to influence the direction the tesla moves in, mostly preventing random walks through space. This is a little costly, but so much cheaper then constant lighting/atmos/explosions it's more then cheap enough.
Adds a flag to zaps that allows them to generate power, applies it to only supermatter zaps.
The tesla emits light, but it's white, which seems like a missed opportunity. I made it industrial blue.
Prevents the tesla from loosing energy, cause that's just kinda sad yaknow?

## Why It's Good For The Game

REEEEEEEEEEEEEEEEEEEEEEEEEadd tesla

Also it fills its original niche of deleting the station without making massive lag better now. I wanna see people fry. 

## Changelog
:cl:
add: Readds the tesla with some changes, it's more about killing machines and people then the hull now. Oh and it makes pretty blue light now. Power delam your local supermatter crystal today
add: You feel the universe shift, tesla code has changed.
fix: Most zaps no longer generate power, excepting the sm zaps.
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/door.dm [hunk 1]
```diff
@@ -428,4 +428,9 @@
 	if(. && !(machine_stat & NOPOWER))
 		autoclose_in(DOOR_CLOSE_WAIT)
 
+/obj/machinery/door/zap_act(power, zap_flags)
+	zap_flags &= ~ZAP_OBJ_DAMAGE
```
Comment (ATH1909) [inline code/game/machinery/doors/door.dm:432]: will this make the zaps chained off of a door also not damage objects?

Comment (LemonInTheDark) [reply code/game/machinery/doors/door.dm:432]: no, zap flags is a bitflag, so changing it in zap_act does nothing. This wasn't always the case, but it is now.

**Comment by optimumtact** [comment]
image

**Comment by Cyberboss** [comment]
image

**Comment by LemonInTheDark** [comment]
Help the headcoders are shitposting again

**Comment by XDTM** [comment]
isn't the point of removing other engines to make power easier to balance? If we readded them before doing it we'd be back where we started

**Comment by LemonInTheDark** [comment]
It's incapable of generating power and is only available through a sm delam. I say re-adds because it was totally ripped out in crit's pr.

**Comment by Dennok** [comment]
Dont see any problems in balancing ROUNDSTART power. 
Problems appears when spacemans aquirre power of science cheated additional collectors, upgraded parts, and whole engines. 
You can perfectly balance roundstart engine but what you do when spacemans build second engine, or third? (without completely removing of ability to build).

**Comment by LemonInTheDark** [comment]
That's true, if we keep our current methods of generating power it can never have a true cap, just an effective one. Goof was talking about diminishing returns which I think might help our case.

**Comment by GuyonBroadway** [comment]
Really there should be a more accessible countermeasure to engine releases that enable quick thinking engineers to minimize damage but that genuine traitors can sabotage.

Goon as the anti-singularity rocket launcher. A very bulky specialized device  it is impractical for an engineer to carry around for safekeeping so leaving it in the storage area is best but antags can dispose of it before singuloosing meaning someone has to go get another one shipped from cargo or whatever. 

Strategies, counter stratergies and greater player agency and whatnot.

**Comment by TemporalOroboros** [comment]
If you're worried about engineering building overpowered engine setups, why not take a page from goon and add powernet overloading?

**Comment by GuyonBroadway** [comment]
True but having an easy to access roundstart solution would mitigate some of the "SINGLOOOSE, CALL IT CALL IT CALL IT" scenarios. Gives to a chance to nip it in the bud before it ruins the round for good.

Also +1 to the powernet idea, it gives antags some fun options too.

**Comment by TheMidnightRose** [comment]
Somewhere in powernet code, theres a hard limit on it that prevents more damage than hard crit when shocking if the powernet is directly wired to the engine. Change that part of the code and the true terror of engineering is to assistants exploding themselves trying to hack.

**Comment by LemonInTheDark** [comment]
> 
> 
> Somewhere in powernet code, theres a hard limit on it that prevents more damage than hard crit when shocking if the powernet is directly wired to the engine. Change that part of the code and the true terror of engineering is to assistants exploding themselves trying to hack.

it's capped at 195 in shock code. don't tempt me

**Comment by LemonInTheDark** [comment]
As an overall point, I'm not really interested in debating ways to solve power here. that's most likely better suited to the forum thread or discord. I'm more interested in getting the tesla ball back in the game in a sane/cheap way while still keeping it threatening

**Comment by TheSilverNuke** [comment]
Yeah revenants do that, but on Goon if you hotwire the engine to the power grid any time someone walks by a light then it'd shock them. And that's not a one time ability but a constant as long as the grid has excess power.

**Comment by JenqaDev** [comment]
i love you lemonman

**Comment by PKPenguin321** [comment]
Could they still do non-breaching explosions?

**Comment by LemonInTheDark** [comment]
I don't think that's a good idea, explosions aren't the cheapest things on their own. We'll see how destructive it ends up being

**Comment by LemonInTheDark** [comment]
Damn it

**Comment by optimumtact** [comment]
what's it waiting for?

**Comment by LemonInTheDark** [comment]
This point nothing. I'd like to see how it does on the servers, on my local it's getting SSobj, the process that the test runs on, to 100ms, but my local is fucking trash. If it needs to be tweaked further I can do that.

**Comment by cacogen** [comment]
now just add the singulo back as a delam and all is well

**Comment by LemonInTheDark** [comment]
I don't know how to tell you this, but the singulo was never removed from the sm delams

**Comment by cacogen** [comment]
I need to play more.

**Pull request merged**